src/Core/src/Handlers/Image/ImageHandler.Windows.cs

```diff
@@ -8,33 +9,152 @@ namespace Microsoft.Maui.Handlers
 {
 	public partial class ImageHandler : ViewHandler<IImage, Image>
 	{
+		/// <inheritdoc/>
 		protected override Image CreatePlatformView() => new Image();
 
+		/// <inheritdoc/>
 		protected override void DisconnectHandler(Image platformView)
 		{
 			base.DisconnectHandler(platformView);
 			SourceLoader.Reset();
 		}
 
+		/// <inheritdoc/>
 		public override bool NeedsContainer =>
 			VirtualView?.Background != null ||
+			VirtualView?.Aspect == Aspect.AspectFill ||
 			base.NeedsContainer;
 
+		/// <inheritdoc/>
+		protected override void SetupContainer()
+		{
+			base.SetupContainer();
+
+			// VerticalAlignment only works when the child's Height is Auto
+			PlatformView.Height = double.NaN;
+
+			MapHeight(this, VirtualView);
+			MapWidth(this, VirtualView);
+			MapMaximumHeight(this, VirtualView);
+			MapMaximumWidth(this, VirtualView);
+		}
+
+		/// <inheritdoc/>
+		protected override void RemoveContainer()
+		{
+			base.RemoveContainer();
+
+			MapHeight(this, VirtualView);
```

Comment:
why do we need this when removing ?

Reply:
So that the Image gets the values instead of the container. 

But these should be replaced with `UpdateValue()` calls like Matt mentioned [here](https://github.com/dotnet/maui/pull/15122/files#r1312720120).